### PR TITLE
Add Docker setup for Ollama with FastAPI and Caddy integration

### PR DESCRIPTION
**Pull request by ItsSimko**
## Summary by Sourcery

Add a Docker Compose setup for integrating Ollama with FastAPI and Caddy, providing GPU support and HTTPS access. Implement API key management and a FastAPI proxy server with rate limiting and authentication.

New Features:
- Introduce a Docker Compose setup for running Ollama with a FastAPI wrapper and Caddy reverse proxy, enabling GPU support and HTTPS access.

Enhancements:
- Implement an API key management module for secure API key generation, storage, and validation in FastAPI applications.
- Add a FastAPI proxy server for Ollama API with rate limiting and authentication capabilities.

Documentation:
- Update README with detailed instructions on setting up and using the Docker Compose configuration for Ollama, FastAPI, and Caddy.

### CODE REVIEW

**Comment by ItsSimko** [comment]
@sourcery-ai review

File: api_wrapper.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+app = FastAPI()
+
+# Initialize API key management
+API_KEY_MANAGER = APIKeyManager()
+
+# Print API key information and security warnings
+print("\n" + "="*50)
+print(" " * 5 + "⚠️ IMPORTANT: API Key Information ⚠️" + " " * 5)
+print("="*50)
+print("\n" + " " * 3 + f" Session API Key: {API_KEY_MANAGER.api_key} " + "\n")
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply api_wrapper.py:27]: I think this OK to let the user know their API key?

File: api_wrapper.py [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+        return JSONResponse(
+            content=response.json(),
+            status_code=response.status_code
+        )
+    except requests.exceptions.JSONDecodeError:
+        return JSONResponse(
+            content=response.text,
+            status_code=response.status_code
+        )
+    except Exception as e:
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply api_wrapper.py:125]: I think OK for now...

File: api_wrapper.py [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+        response = await call_next(request)
+        return response
+    except RateLimitExceeded:
+        return PlainTextResponse(
+            "Rate limit exceeded. Try again later.",
+            status_code=429
+        )
+
+@app.api_route("/{path:path}", methods=["GET", "POST"], dependencies=[Depends(API_KEY_MANAGER.verify_api_key)])
+@limiter.limit("1/second")
```
[the remark this answers is not in the document]
Comment (ItsSimko) [reply api_wrapper.py:72]: I think 1 per 1 second is okay

**Review by yingbull — APPROVED** [review]

**Pull request merged**